esp-hal/src/spi/master.rs

```diff
@@ -1107,31 +1110,31 @@ where
         self
     }
 
-    #[procmacros::enable_doc_switch]
+    #[doc_replace(
+        "max_frequency" => {
+            cfg(esp32h2) => " 48MHz",
+            _ => " 80MHz",
+        }
+    )]
     /// Change the bus configuration.
     ///
     /// # Errors
     ///
     /// If frequency passed in config exceeds
-    #[doc_switch(
-        cfg(esp32h2) => " 48MHz",
-        _ => " 80MHz",
-    )]
-    /// or is below 70kHz,
-    /// [`ConfigError::UnsupportedFrequency`] error will be returned.
+    /// # {max_frequency}
+    /// or is below 70kHz, [`ConfigError::UnsupportedFrequency`] error will be returned.
```

Comment:
This isn't ideal, but it renders as expected. Replacing in the middle of a line would be preferrable, but also more complicated to implement.